http-filter-example/http_filter.cc

```diff
@@ -7,19 +7,24 @@
 namespace Envoy {
 namespace Http {
 
-HttpSampleDecoderFilter::HttpSampleDecoderFilter() {}
+HttpSampleDecoderFilterConfig::HttpSampleDecoderFilterConfig(
+    const sample::Decoder& proto_config)
+    : key_(proto_config.key()), val_(proto_config.val()) {}
+
+HttpSampleDecoderFilter::HttpSampleDecoderFilter(HttpSampleDecoderFilterConfigSharedPtr config)
+    : config_(config) {}
 
 HttpSampleDecoderFilter::~HttpSampleDecoderFilter() {}
 
 void HttpSampleDecoderFilter::onDestroy() {}
 
 const LowerCaseString& HttpSampleDecoderFilter::headerKey() {
-  static LowerCaseString* key = new LowerCaseString("via");
+  static LowerCaseString* key = new LowerCaseString(config_->key());
```

Comment:
This should now be per-object rather than static, since different instances of the filter might use different key/values.